framework/include/userobject/ElementLoopUserObject.h

```diff
@@ -0,0 +1,89 @@
+#ifndef ELEMENTLOOPUSEROBJECT_H
+#define ELEMENTLOOPUSEROBJECT_H
+
+#include "GeneralUserObject.h"
+#include "ThreadedElementLoop.h"
+#include "Coupleable.h"
+#include "MooseVariableInterface.h"
+#include "ScalarCoupleable.h"
+#include "MooseVariableDependencyInterface.h"
+#include "ZeroInterface.h"
+#include "MooseVariable.h"
+#include "SubProblem.h"
+#include "NonlinearSystem.h"
+#include "Assembly.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/parallel_algebra.h"
+
+// Forward Declarations
+class ElementLoopUserObject;
+
+namespace libMesh
+{
+class Elem;
+class QBase;
+}
+
+template<>
+InputParameters validParams<ElementLoopUserObject>();
+
+/**
+ * This will loop over elements and do things
+ *
+ * !!! EXPERIMENTAL !!! Just to see what it takes to do such a thing. Not meant for real use ;-)
+ */
+class ElementLoopUserObject :
+    public GeneralUserObject,
+    public ThreadedElementLoop<ConstElemRange>,
```

Comment:
Ok, that does make sense. I'm still not convinced though that having the user object _be_ the thread loop object is the way to go here.